### PR TITLE
refactor: specify timeout for HTTP calls in `run -p/--publish` test

### PR DESCRIPTION
**Pull request by vsiravar**
Issue #, if available:

*Description of changes:*
Code review fix for https://github.com/runfinch/common-tests/pull/39
*Testing done:*

Yes

- [X] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ningziwen** [comment]
I think the title should describe the actual things you are fixing.

File: fnet/dial.go [hunk 1]
```diff
@@ -14,16 +14,21 @@ import (
...
 		// #nosec G107 // it does not matter if url is not a constant for testing.
-		resp, err := http.Get(url)
+		resp, err := client.Get(url)
 		if err != nil {
 			time.Sleep(retryInterval)
 			continue
 		}
 
+		//nolint:errcheck // not required to handle error for testing if resp.Body.Close() returns error
+		defer resp.Body.Close()
```
Comment (davidhsingyuchen) [inline fnet/dial.go:30]:
```suggestion
defer func() { gomega.Expect(resp.Body.Close()).To(gomega.Succeed()) }()
```

**Review by davidhsingyuchen — COMMENTED** [review]
nit: maybe this should be` refactor:` instead of `fix:`?

File: fnet/dial.go (hunk 1 above)
Comment (vsiravar) [reply fnet/dial.go]: Makes sense, made the change.

**Review by davidhsingyuchen — APPROVED** [review]

**Pull request merged**